.github/workflows/ci.yml

```diff
@@ -47,7 +47,20 @@ jobs:
               run: bash -x ./scripts/ubuntu_wayland_setup
             - name: Run Tests
               run: |
+                set +e
                 [ "$(grep -c -P '\t' CHANGELOG)" = "0" ]
+                uv tool install .
+                uv tool install --reinstall '.[wayland]'
+                uv run --python=${{ matrix.python-version }} ./libqtile/backend/wayland/cffi/build.py
```

Comment:
don't you need to do this before you install?

Reply:
hmm, i think you're right. let me try that.

Reply:
Try what locally? A version installed with `uv`? I have been running it since before the patch landed.

But a question about your environment there: there is no X server running, so it makes sense that there's no qtile socket to connect to. You probably need xvfb or Xephyr or something like the test suite uses.

Reply:
Wayland is the issue I am looking at, which why I sent the exact link :)

X11 is working fine in the other CI, not sure why it's not working here, and I did start Xvfb here as well. Anyway, any chance you can look at the wayland run on 3.12 I shared above?

Reply:
I see:
```
Traceback (most recent call last):
  File "/home/runner/work/qtile/qtile/./libqtile/backend/wayland/cffi/build.py", line 3, in <module>
    import wlroots.ffi_build as wlr
ModuleNotFoundError: No module named 'wlroots'
```

I think you have not called `uv sync --all-extras` before doing the backend build, and it just fails, but continues on because the script is `set +e` and not `set -e`.

Reply:
Oops, completely glanced over that. Sorry!

I did do that in the other CI (i keep saying this, haha), look: https://github.com/ervinpopescu/qtile-cmd-client/actions/runs/16985241914/job/48158611073#step:7:1

Anyway, let me try Wayland locally, because this seems fishy.

Reply:
I didn't have time and an available device to try it on, so here's v0.33.0 working perfectly fine: https://github.com/ervinpopescu/qtile-cmd-client/actions/runs/16998613390/job/48195264787

Reply:
```
$ uv run --with setuptools --directory=$(uv tool dir)/qtile/lib/python3.13/site-packages/ $(uv tool dir)/qtile/lib/python3.13/site-packages/libqtile/backend/wayland/cffi/build.py
```

This fixed it locally. It turns out `uv run` will build it under the current project, but not actually move the `_ffi.*` files to `$(uv tool dir)`.

If `--with setuptools` is not provided:
```
Exception: This CFFI feature requires setuptools on Python >= 3.12. The setuptools module is missing or non-functional.
```

I'm not sure how I can use qtile's setuptools with `uv run`, I'm going to take a look.